src/chkstat.cpp

```diff
@@ -612,8 +639,83 @@ bool Chkstat::parseProfile(const std::string &path)
             continue;
         }
 
-        const auto &entry = addProfileEntry(location, user, group, mode_int);
-        last_key = entry.file;
+        std::vector<std::string> expansions;
+        if (!expandProfilePaths(location, expansions) )
+        {
+            badProfileLine(path, linenr, "bad variable expansions");
+            continue;
+        }
+
+        for (const auto &exp_path: expansions)
+        {
+            addProfileEntry(exp_path, user, group, mode_int);
+            // remember the most recently added entries to allow potential '+'
+            // capability lines to be applied to them.
+            active_keys.push_back(exp_path);
+        }
+    }
+
+    return true;
+}
+
+bool Chkstat::expandProfilePaths(const std::string &path, std::vector<std::string> &expansions)
+{
+    std::stringstream ss;
+    ss.str(path);
+    std::string part;
+
+    // the initial entry
+    expansions.clear();
+    expansions.push_back("");
+
+    // process each path component.
+    //
+    // we support variables only as individual
+    // path components i.e. something like %{myvar}stuff/suffix is not
+    // allowed, only %{myvar}/suffix.
+    //
+    // multiple variable components in the same path are supported
```

Comment:
Implicitly this parsing means we don't support slashes in variable *names*. That's perfectly fine. I'm just wondering if we should make this more explicit.

Reply:
I think no special characters should be allowed in variable names. alphanumerical only. plus underscore maybe. I can add checking logic for that.

Reply:
Also makes sense :)